Eigen/src/Geometry/EulerAngles.h

```diff
@@ -55,7 +55,12 @@ MatrixBase<Derived>::eulerAngles(Index a0, Index a1, Index a2) const
     res[0] = atan2(coeff(j,i), coeff(k,i));
     if((odd && res[0]<Scalar(0)) || ((!odd) && res[0]>Scalar(0)))
     {
-      res[0] = (res[0] > Scalar(0)) ? res[0] - Scalar(EIGEN_PI) : res[0] + Scalar(EIGEN_PI);
+      if(res[0] > Scalar(0)) {
+        res[0] -= Scalar(EIGEN_PI);
+      }
+      else {
+        res[0] += Scalar(EIGEN_PI);
+      }
```

Comment:
Good catch. But just in case you did not notice the line following this if-else statement, it should also handle res[1] for both `if` case and `else if` case, so it would look like
```
if ((odd && res[0]<Scalar(0)){ 
  res[0] += Scalar(EIGEN_PI);
  res[1] = atan2(-coeff(i,k), -c2);
}
else if (((!odd) && res[0]>Scalar(0))) {
  res[0] -= Scalar(EIGEN_PI);
  res[1] = atan2(-coeff(i,k), -c2);
}
```
while in the previous case it handles `res[1]` after the if-else statement. Do you think it is simpler to handle `res[1]` after the if-else statement, as already in the commit, or put the `res[1]` inside the if-else statement.